File: tests/test2.c
#include <stdio.h> 
 
void scopeinner(){
int ten;
ten = 10;
printf(ten);
if(ten == 10){
ten = 1;
}
}

int summation(int num){
int summation;
if(num == 1){
summation = 1;
} else {
summation = 2;
}
return summation;
}


int main(void){
int a;
a = 20;
printf(a + 1 * 2);
scopeinner();
a = a + summation(10);
printf(a);
return 0; 
}

File: tests/test2.pas
program test2;
var
    A : integer;

procedure ScopeInner;
var ten : integer;
begin
    ten := 10;
    write(ten);
    if ten = 10 then
    begin
        ten := 1;
    end;
end;

function Summation(num : integer) : integer;
begin
    if num = 1 then
    begin
        Summation := 1;
    end
    else
    begin
        Summation := 2;
    end;
end;

begin
    A := 20;
    write(A + 1 * 2);
    ScopeInner;
    A := a + Summation( 10 );
    write(a);
end.